warehouse/query-core/src/main/java/datawave/core/iterators/TermFrequencyIterator.java

```diff
@@ -191,36 +190,28 @@ protected void findTop() throws IOException {
                 break;
             }
             
-            if (count > MAX_SCAN_BEFORE_SEEK) {
+            tfKey.parse(k);
```

Comment:
Much cleaner, however I wish we could avoid a new class member and instead pass this down into the functions.  It just makes it harder to verify that the members are filled/set wherever they are used.

Reply:
I would tend to agree with you, however in this case everything passes through the hasTop() method so there's a single entry point for new keys.